File: src/app/api/groq-models/route.py
import requests
import os

api_key ="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"
url = "https://api.groq.com/openai/v1/models"

headers = {
    "Authorization": f"Bearer {api_key}",
    "Content-Type": "application/json"
}

response = requests.get(url, headers=headers)
data = response.json()  # Parse JSON from response

    # Extract the models
models = [{"id": model["id"], "owned_by": model["owned_by"]} for model in data["data"]]

print(models)

File: src/app/api/groq-models/route.ts
import Groq from "groq-sdk";
import { NextResponse } from "next/server";

//Initialize the Groq client
const client = new Groq({
  apiKey: process.env["GROQ_API_KEY"],
});

interface Model {
  id: string;
  owned_by: string;
}

export async function GET() {
  try {
    const response = await client.models.list();
    const models: Model[] = response.data.map(
      (model: { id: string; owned_by: string }) => ({
        id: model.id,
        owned_by: model.owned_by,
      })
    );
    console.log(models);
    return NextResponse.json({ models: models });
  } catch (error) {
    console.error("Error fetching models:", error);
  }
}
